### DISCUSSION TITLE
Post-Upgrade tasks not running for existing PRs

### DISCUSSION THREAD

**Discussion by jochenberger** [renovatebot/renovate#12337] [Request Help]
Hi,

I'm using the self-hosted variant (28.10.2) with a bitbucket server.
I added some post-upgrade tasks to the renovate configuration when some PRs were already created. When the bot rebased the branches after I pushed the config change, the PRs were not updated to reflect the configuration change, i.e. the post-upgrade tasks were not being run. There's nothing suspicious in the logs. Am I doing something wrong or am I asking too much?

**Comment by viceice** [comment]
Do you configure https://docs.renovatebot.com/self-hosted-configuration/#allowedpostupgradecommands ?

**Reply by jochenberger** [reply]
No, I didn't. I only just noticed that myself. I think that a warning should be logged in this case. Should I open an issue?

**Reply by jochenberger** [reply]
Now it tries to execute the tasks for NEW PRs, but not for existing ones, i.e. PRs that were created *before* the `postUpgradeTasks` were added.

**Reply by viceice** [reply]
yes, you need to rebase those to get the tasks run on existing pr's

**Reply by jochenberger** [reply]
Yes, I'd expect the bot to do that when it rebases anyway because of the repo's settings (which lead to `rebaseWhen: behind-base-branch`).

**Reply by rarkins** [reply]
Yes, it should run them on any rebase

**Reply by jochenberger** [reply]
Yes, it should. But it doesn't.

**Comment by jochenberger** [comment]
Now, for new PRs, the execution fails with
```
"artifactErrors": [
         {
           "lockFile": "build.gradle",
           "stderr": "Command failed: ./gradlew generate\n"
         }
       ]
```
Unfortunately, I can't tell what's happening from the error message.

**Reply by jochenberger** [reply]
Maybe it's because the `slim` distribution has no JDK installed and it fails to run Gradle. Just guessing though. I'll try the regular image.

**Reply by viceice** [reply]
yes, the slim image only comtains git, node and renovate, so post-upgrade tasks won't work if they depend on other binaries.

**Reply by jochenberger** [reply]
Yes, that's it.
I wonder if the error message could be improved in that case. I guess the Gradle wrapper script outputs a helpful error message if no JDK is available.

**Reply by viceice** [reply]
you need to check the logs, there you should find more details
